modules/plugins/jboss-as-7/src/main/java/org/rhq/modules/plugins/jbossas7/JBossProduct.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * RHQ Management Platform
+ * Copyright (C) 2005-2015 Red Hat, Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+package org.rhq.modules.plugins.jbossas7;
+
+import java.io.File;
+import java.io.FileFilter;
+
+/**
+ * this class tends to be replacement of {@link JBossProductType} which was designed as enum and we're hitting enum
+ * limitations in AS7-based plugins
+ * @author lzoubek
+ *
+ */
+@SuppressWarnings("deprecation")
+public class JBossProduct {
+
+    private static FileFilter homeDirMatchFilter(final String... substrings) {
+        return new FileFilter() {
+
+            @Override
+            public boolean accept(File pathname) {
+                if (substrings != null) {
+                    String homeDir = pathname.getName();
+                    for (String s : substrings) {
+                        if (s != null && homeDir.contains(s)) {
+                            return true;
+                        }
+                    }
+                }
+                return false;
+            }
+        };
+    }
+
+    public static JBossProduct eap() {
+        return new JBossProduct(JBossProductType.EAP, "eap", homeDirMatchFilter("-eap-"));
+    }
+
+    public static JBossProduct wildfly8() {
+        return new JBossProduct(JBossProductType.WILDFLY8, null, homeDirMatchFilter("wildfly"));
+    }
+
+    public static JBossProduct ispn() {
+        return new JBossProduct(JBossProductType.ISPN, "ispn", homeDirMatchFilter("infinispan-server"));
+    }
+
+    public static JBossProduct jdg() {
+        return new JBossProduct(JBossProductType.JDG, "jdg", homeDirMatchFilter("-jdg-", "datagrid-server"));
+    }
+
+    public static JBossProduct jpp() {
+        return new JBossProduct(JBossProductType.JPP, "jpp", homeDirMatchFilter("-jpp-"));
+    }
+
+    public static JBossProduct soa() {
+        return new JBossProduct(JBossProductType.SOA, "soa", null);
+    }
+
+    public static JBossProduct as() {
+        return new JBossProduct(JBossProductType.AS, null, homeDirMatchFilter("-as-"));
+    }
+
+    public static JBossProduct jdv() {
+        return new JBossProduct(JBossProductType.JDV, "dv", null);
+    }
+
+    public static JBossProduct unknown() {
+        return unknown(null);
+    }
+
```

Comment:
Maybe we can cache all these (in a static final field) to avoid creating instances over and over?